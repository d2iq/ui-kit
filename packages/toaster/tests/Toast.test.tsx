import * as React from "react";
import { shallow, mount } from "enzyme";
import { createSerializer } from "@emotion/jest";
import toJson from "enzyme-to-json";
import { Toaster, Toast } from "../";

expect.addSnapshotSerializer(createSerializer());

describe("Toast", () => {
  it("renders default", () => {
    const component = shallow(
      <Toaster children={[<Toast title="I Am Toast" key={0} id={0} />]} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("renders with description", () => {
    const component = shallow(
      <Toaster
        children={[
          <Toast title="I Am Toast" description="Description" key={0} id={0} />
        ]}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("renders with actions", () => {
    const component = shallow(
      <Toaster
        children={[
          <Toast
            primaryAction={<button>primaryAction</button>}
            secondaryAction={<button>secondaryAction</button>}
            title="I Am Toast"
            key={0}
            id={0}
          />
        ]}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should call the function in the action when clicked", () => {
    const action = jest.fn();
    const component = mount(
      <Toaster
        children={[
          <Toast
            primaryAction={<button onClick={action}>primaryAction</button>}
            title="I Am Toast"
            key={0}
            id={0}
          />
        ]}
      />
    );
    const primaryActionBtn = component.find(`button`);

    expect(action).not.toHaveBeenCalled();
    primaryActionBtn.simulate("click");
    expect(action).toHaveBeenCalled();
  });

  it("calls dismissToast when the dismiss button is clicked", () => {
    const dismissToastSpy = spyOn(Toaster.prototype, "dismissToast");
    const component = mount(
      <Toaster children={[<Toast title="I Am Toast" key={0} id={0} />]} />
    );
    const dismissBtn = component.find('span[role="button"]');
    expect(dismissToastSpy).not.toHaveBeenCalled();
    dismissBtn.simulate("click");
    expect(dismissToastSpy).toHaveBeenCalled();
  });

  it("should remove the Toast from the Toaster `toasts` state when the dismiss button is clicked", () => {
    const component = mount(
      <Toaster children={[<Toast title="I Am Toast" key={0} id={0} />]} />
    );
    const instance = component.find(Toaster).instance() as Toaster;
    const dismissBtn = component.find('span[role="button"]');
    let toastsState = instance.state.toasts || [];

    expect(toastsState.length).toBe(1);

    dismissBtn.simulate("click");
    toastsState = instance.state.toasts || [];

    expect(toastsState.length).toBe(0);
  });
});
