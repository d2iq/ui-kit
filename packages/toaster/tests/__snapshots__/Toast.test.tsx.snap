// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders default 1`] = `
.emotion-0 {
  max-width: 100%;
  margin: 16px;
}

@media (min-width: 900px) {
  .emotion-0 {
    max-width: 20em;
  }
}

@media (min-width: 900px) {
  .emotion-0 {
    margin: 24px;
  }
}

.emotion-1 {
  list-style: none;
  margin-left: 0;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
}

<div
  className="emotion-0"
  data-cy="toaster"
>
  <ol
    aria-live="assertive"
    className="emotion-1"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <TransitionGroup
      childFactory={[Function]}
      component="div"
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={false}
        key="toastWrapper-0"
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={
          Object {
            "enter": 0,
            "exit": 300,
          }
        }
        unmountOnExit={false}
      >
        <Component />
      </Transition>
    </TransitionGroup>
  </ol>
</div>
`;

exports[`Toast renders with actions 1`] = `
.emotion-0 {
  max-width: 100%;
  margin: 16px;
}

@media (min-width: 900px) {
  .emotion-0 {
    max-width: 20em;
  }
}

@media (min-width: 900px) {
  .emotion-0 {
    margin: 24px;
  }
}

.emotion-1 {
  list-style: none;
  margin-left: 0;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
}

<div
  className="emotion-0"
  data-cy="toaster"
>
  <ol
    aria-live="assertive"
    className="emotion-1"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <TransitionGroup
      childFactory={[Function]}
      component="div"
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={false}
        key="toastWrapper-0"
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={
          Object {
            "enter": 0,
            "exit": 300,
          }
        }
        unmountOnExit={false}
      >
        <Component />
      </Transition>
    </TransitionGroup>
  </ol>
</div>
`;

exports[`Toast renders with description 1`] = `
.emotion-0 {
  max-width: 100%;
  margin: 16px;
}

@media (min-width: 900px) {
  .emotion-0 {
    max-width: 20em;
  }
}

@media (min-width: 900px) {
  .emotion-0 {
    margin: 24px;
  }
}

.emotion-1 {
  list-style: none;
  margin-left: 0;
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
}

<div
  className="emotion-0"
  data-cy="toaster"
>
  <ol
    aria-live="assertive"
    className="emotion-1"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <TransitionGroup
      childFactory={[Function]}
      component="div"
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={false}
        key="toastWrapper-0"
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={
          Object {
            "enter": 0,
            "exit": 300,
          }
        }
        unmountOnExit={false}
      >
        <Component />
      </Transition>
    </TransitionGroup>
  </ol>
</div>
`;
